### PR TITLE
Rework console warnings + add defaults + no 3 of same family behaviour + minrankranked and co + maxconnectedgames maxconnectedgamesperuser + fix missing exports

### PR DESCRIPTION
**Pull request by wonderingabout**
Many new changes in this PR, some are important :

general look here : https://github.com/wonderingabout/gtp2ogs/tree/rework-console-minrankranked-and-co 

- removed console warnings for maxmaintime and co "setting not detected"
- instead, using default values hassle-free, large and friendly for rescue value purposes
- change behaviour if 3 arguments of same family are used : for arguments that have a general+ranked+unranked possible values, new behaviour is now, for example for `--maxmaintime` :  general value (default or manually inputted by bot admin) will be taken into account only if no specific value is used (ranked and/or unranked); else bot admin will have to set up both ranked and unranked values desired and general value will not be taken into account (added notes about that on readme)
- added `--minrankranked` and co : if bot admin wants to specify different minrank and maxrank for ranked and unranked games
- renamed `--maxactivegames` and `--maxtotalgames` to the clearer and more powerful `--maxconnectedgames` and `--maxconnectedgamesperuser` : these names are clearer but also more powerful : in the future, someone may want to add a maxtotalgames per bot and/or per user which is something different (max number of total games in the list on ogs profile, see gnugo page for example : it has a lot of games in the list but gnugo is not connected to almost any of them, bot admins may want to have the option to regulate that in the future) @lemonsqueeze 
- updated readme and ISSA to take all that into account

@roy7 @Dorus @windo @lemonsqueeze

### CODE REVIEW

**Comment by wonderingabout** [comment]
update : currently not working for some reason, most likely when i renamed maxactivegames to maxtotalgames

will debug next time

**Comment by wonderingabout** [comment]
tested to work on beta ogs ingame

**Comment by roy7** [comment]
> correspondence games are currently included in the connected games count if you use `--persist`

I don't think this is how it works? I believe every game in progress will be in the connected games list, whether it is your turn or not. --persist won't change that. --persist would change the # of engines in memory/active though, if you go through the connected games list to count how many have valid game states? I'm just going from memory here.

File: README.md [hunk 1]
```diff
@@ -241,10 +250,10 @@ are banned from playing unranked game
 byoyomi,simple,canadian,absolute,none)
 
   ```--minmaintime```  or ```--0mt``` Minimum seconds of main time (rejects time 
-control simple and none)
+control simple and none) (default 60)
 
   ```--maxmaintime```  or ```--1mt``` Maximum seconds of main time (rejects time 
-control simple and none)
+control simple and none) (default 1800)
```
Comment (roy7) [inline README.md]: This is only 30 minutes. How about making default max main time 2 or 4 hours. The default on max time is more of a "prevent unintended unlimited time". If people want less than a high amount they can lower in settings. 30 minutes feels too low for default max, since often people do in fact play bots with longer clocks?

Comment (wonderingabout) [reply README.md]: i see, this is why i wanted to make separate time settings for everygame speed : 
- `--maxmaintimeblitzlive 60*60`
- `--maxmaintimecorr 60*60*24*7`

i think every speed should have its own time speed @roy7 
more than 1 hour main time is not a live game anymore (blitz and live are kidna same so same category)
if main time is more than 1 hour, i suggest the human player should play a corr game instead

as you said, bot admin is free to increase to `--maxmaintimeblitzlive 60*60*4` if want @roy7 
i think this makes more sense 

see #125 
also, with a config file we'd be able to do math like that automatically which is also a nice advantage

to sum up, i can increase to 1 hour, but more than that should be a corr game by default, unless bot admin wants to manually change that
(after main time there are still periods : default is 1 hour + 20 x 2 min if i remember correctly, theres plenty of time for a live game, much more than on foxwq or tygem for example)

**Comment by wonderingabout** [comment]
> > correspondence games are currently included in the connected games count if you use `--persist`
> 
> I don't think this is how it works? I believe every game in progress will be in the connected games list, whether it is your turn or not. --persist won't change that. --persist would change the # of engines in memory/active though, if you go through the connected games list to count how many have valid game states? I'm just going from memory here.

maybe i confused `--persist` with corr settings, because i remember there was an issue here in #59

File: README.md (hunk 1 above)
Comment (Dorus) [reply README.md]: I agree with Roy here. Default should be something large like 2-4 hours. If bot admins find that too much, they can decrease it.

**Comment by wonderingabout** [comment]
so to sum up i'll increase maxmaintimelive to 3600

and more importantly, i'll enhance the UHMAEAT to support separate blitz,live, and corr timesettings (we'll have 3 UHMEAT for maintime, and 3 for periodtime)

for example : 

maxmaintimeblitz 60*5
maxmaintimelive 60*60
maxmaintimecorr 60*60*24*7

maxperiodtimebltz 30
maxperiodtimelive 60*2
maxperiodtimecorr 60*60*24*3

@Dorus @roy7

File: README.md (hunk 1 above)
Comment (wonderingabout) [reply README.md]: ok, i'll increase it to 2 hours then, but remember that there is still periodtime on top of that

and with maxmaintime 2 hours, it means corr games are not possible, all the more reason to add a specific setting for blitz, live corr, see my messages in https://github.com/online-go/gtp2ogs/pull/123#issuecomment-462112142

**Comment by wonderingabout** [comment]
reverted to 7200 default in #124 

let's continue the discussion there :+1:
